Repo: prettier-update.yml errors on git push: "fatal: You are not currently on a branch."

### Suggestion

From #7425 -> https://github.com/typescript-eslint/typescript-eslint/actions/runs/5772703419/job/15647950043?pr=7425:

```plaintext
...
tsconfig.eslint.json 1ms
Done in 24.80s.
[detached HEAD 560d50c] chore: update formatting after prettier upgrade
 15 files changed, 66 insertions(+), 25 deletions(-)
fatal: You are not currently on a branch.
To push the history leading to the current (detached HEAD)
state now, use

    git push origin HEAD:<name-of-remote-branch>

Error: Process completed with exit code 128.
```

--- a/.github/workflows/prettier-update.yml
+++ b/.github/workflows/prettier-update.yml
@@ -21,6 +21,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          persist-credentials: false
 
       - name: Check if prettier was changed as part of the latest commit on the PR
         id: prettier-package-check

